File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int N = Integer.parseInt(br.readLine());
        Deque<Integer> que = new LinkedList<>();
        for(int i = 0; i < N; i++) {
            String S = br.readLine();
            if(S.equals("front")) {
                sb.append(que.isEmpty() ? -1 : que.peek()).append("\n");
            } else if(S.equals("back")) {
                sb.append(que.isEmpty() ? -1 : que.peekLast()).append("\n");
            } else if(S.equals("size")) {
                sb.append(que.size()).append("\n");
            } else if(S.equals("empty")) {
                sb.append(que.isEmpty() ? 1 : 0).append("\n");
            } else if(S.equals("pop")) {
                sb.append(que.isEmpty() ? -1 : que.poll()).append("\n");;
            } else {
                que.add(Integer.parseInt(S.split(" ")[1]));
            }
        }
        System.out.println(sb.toString().trim());
    }
}

File: 백준/Silver/10845. 큐/큐.js
class Deque {
  constructor() {
    this.arr = [];
    this.head = 0;
    this.tail = 0;
  }

  peekFirst() {
    return this.arr.length === 0 ? -1 : this.arr[0];
  }

  peekLast() {
    return this.arr.length === 0 ? -1 : this.arr[this.arr.length - 1];
  }

  size() {
    return this.arr.length;
  }

  isEmpty() {
    return this.arr.length === 0 ? 1 : 0;
  }

  pop() {
    return this.arr.length === 0 ? -1 : this.arr.shift();
  }

  push(num) {
    this.arr.push(num)
  }
}

let [N, ...input] = require('fs').readFileSync('/dev/stdin').toString().split('\n').map(line => line.trim());
N = Number(N);
let deque = new Deque();
const ans = [];
for(let i = 0; i < N; i++) {
  if(input[i] === 'front') {
    ans.push(deque.peekFirst());
  } else if(input[i] === 'back') {
    ans.push(deque.peekLast());
  } else if(input[i] === 'size') {
    ans.push(deque.size());
  } else if(input[i] === 'empty') {
    ans.push(deque.isEmpty());
  } else if(input[i] === 'pop') {
    ans.push(deque.pop());
  } else {
    deque.push(Number(input[i].split(' ')[1]));
  }
}
console.log(ans.join('\n'));